### PR TITLE
feat(parquet): process preview_offset and preview_nrows [TCTC-9803]

### PR DESCRIPTION
**Pull request by Fanaen**
## Change Summary

- Add a new reader function for parquet (it was previously routed directly into `pd.read_parquet`)
- Use instead `pyarrow` + `.to_pandas()` to get efficient lazy-loading.
- Handle `previews_nrows`, `previews_offset` and `columns`

## Checklist

* [x] Unit tests for the changes exist
* [ ] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable

### CODE REVIEW

File: peakina/readers/parquet.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    FilePathOrBuffer = str | bytes | PathLike[str] | PathLike[bytes]
+
+
+@wraps(pd.read_parquet)
+def read_parquet(
+    path_or_buf: "FilePathOrBuffer",
+    preview_offset: int = 0,
+    preview_nrows: int | None = None,
+    columns: list[int] | None = None,
+) -> pd.DataFrame:
```
Comment (lukapeschke) [inline peakina/readers/parquet.py]:
```suggestion
    columns: list[int] | None = None,
    **kwargs
) -> pd.DataFrame:
```

I believe this should be added to ignore extra args, which can happen if the type of a datasource is modified

File: peakina/readers/parquet.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import pandas as pd
+import pyarrow.dataset as ds
+
+if TYPE_CHECKING:
+    from os import PathLike
+
+    FilePathOrBuffer = str | bytes | PathLike[str] | PathLike[bytes]
+
+
+@wraps(pd.read_parquet)
```
Comment (lukapeschke) [inline peakina/readers/parquet.py]: Not true

File: peakina/readers/parquet.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    preview_nrows: int | None = None,
+    columns: list[int] | None = None,
+) -> pd.DataFrame:
+    dataset = ds.dataset(source=path_or_buf, format="parquet")
+
+    if preview_nrows is not None:
+        indices = range(preview_offset, preview_offset + preview_nrows)
+        table = dataset.take(indices=indices, columns=columns)
+    else:
+        table = dataset.to_table(columns=columns)
```
Comment (lukapeschke) [inline peakina/readers/parquet.py:34]: Is the offset is ignored here ?

File: tests/readers/test_parquet.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    ds = DataSource(
+        path("fixture.parquet"),
+        reader_kwargs={"preview_nrows": 2, "columns": ["Date", "Country"]},
+    )
+    assert ds.get_df().shape == (2, 2)
+    assert ds.get_df().equals(
+        pd.DataFrame({"Date": ["29/01/1900", "31/07/1900"], "Country": ["Australia", "Croatia"]})
+    )
+
+    # preview with `offset` and `nrows`
```
Comment (lukapeschke) [inline tests/readers/test_parquet.py:21]: Could you please add a test with  `offset` only ?

**Review by lukapeschke — CHANGES_REQUESTED** [review]
Thanks!

File: peakina/readers/parquet.py (hunk 3 above)
Comment (Fanaen) [reply peakina/readers/parquet.py:34]: Yes. I assumed the `preview_offset` is not relevant is not relevant if there is no preview at all. It should skip rows ?

File: peakina/readers/parquet.py (hunk 2 above)
Comment (Fanaen) [reply peakina/readers/parquet.py]: [5df7f78](https://github.com/ToucanToco/peakina/pull/910/commits/5df7f78c94564a795dae3afc71bc1e9dcd3faf0e)

File: tests/readers/test_parquet.py (hunk 4 above)
Comment (Fanaen) [reply tests/readers/test_parquet.py:21]: [0d22717](https://github.com/ToucanToco/peakina/pull/910/commits/0d22717187772532ad90e6080c910ab1c15fa70c)

File: peakina/readers/parquet.py (hunk 3 above)
Comment (Fanaen) [reply peakina/readers/parquet.py:34]: [0d22717](https://github.com/ToucanToco/peakina/pull/910/commits/0d22717187772532ad90e6080c910ab1c15fa70c)

File: peakina/readers/parquet.py (hunk 1 above)
Comment (Fanaen) [reply peakina/readers/parquet.py]: Ok, I applied it. It triggers a lint test `Function is missing a type annotation`. How do I deal with this? 
That's completely uncharted territory for me.

[f8c5b97](https://github.com/ToucanToco/peakina/pull/910/commits/f8c5b9761762319a399671dfded346190f672de8) [3a6db93](https://github.com/ToucanToco/peakina/pull/910/commits/3a6db93e412cfd74538b5408638b0bb0e1065c92)

Comment (Fanaen) [reply peakina/readers/parquet.py]: [fe3ea55](https://github.com/ToucanToco/peakina/pull/910/commits/fe3ea55e94e59d81cf584377ee40409c158e0dbb)

**Review by lukapeschke — APPROVED** [review]
:ok_hand:

**Pull request merged**